### ISSUE TITLE
Training Parameters: Nadam optimizer, learning rate

### ISSUE THREAD

**Issue by CodyKurpanek** [ameraner/dsen2-cr#12]
Hello @ameraner ,
You have a wonderful project here, and I am interested in replicating the results that your team has achieved as a first step.

In the paper I see that you are using an adam optimizer with momentum of 0.5. 
Does that correspond with the three parameters beta_1, beta_2, and epsilon here? 
https://github.com/ameraner/dsen2-cr/blob/59e81ae0d013bcae992d078e92a1175cc4cbf257/Code/dsen2cr_main.py#L80
Do you have any advice on how to set those three parameters; should I keep them as they are?

Also, in the paper I see a multiplicative factor used to decay the learning rate.
I was thinking of training the first 5 epochs first with schedule_decay=0, then to use --resume for each epoch afterwards with schedule_decay=0, but multiplying the multiplicative factor to the previous lr. Is that similar to the process you used or is there a better way to do that?

Thank you

**Comment by ameraner** [comment]
Hi @CodyKurpanek , 
thank you for your questions and sorry for the late answer! 

I believe there is a small misunderstanding: the section of the paper mentioning a momentum of 0.5 is referring to the cGAN model we used to compare the performance of DSen2-CR, which is unfortunately not part of the code base of this repo. This model was using plain Adam with pre-defined momentum.

In the case of DSen2-CR, I was using Adam with the Nesterov momentum (`Nadam`) . I did not touch its parameters other than the learning rate `lr`, leaving them with the default values. For the learning rate I ran a search algorithm to find the best value (which turned out to be 7e-5). I also ran some experiments with decaying learning rates using different approaches, but I did not find significant improvements on the validation runs, so I did not include this in the final model.
